internal/gps/pkgtree/pkgtree.go

```diff
@@ -152,18 +154,23 @@ func ListPackages(fileRoot, importRoot string) (PackageTree, error) {
 			}
 		}
 
+		if pkg.CommentPath != "" && !strings.HasPrefix(pkg.CommentPath, importRoot) {
+			return &NonCanonicalImportRoot{
```

Comment:
this needs softening a bit, as right now, the only cases where we return an actual error from `ListPackages()` itself (which is what this would result in) are more physical than logical - e.g., we can't actually read the filesystem. the rest of the API is built around that kind of assumption, too.

so, instead of returning the error directly, let's make this an error that we put in the `PackageOrErr.Err` slot on each package where we encounter the problem.

Reply:
I presume and equivalent change should be made to prevent `ConflictingImportComments` from leaking as well?